ko/lessons/specifics/ecto.md

```diff
@@ -0,0 +1,328 @@
+  hostname: "localhost"
+```
+
+## Mix 태스크
+
+Ecto에는 데이터베이스와 작업할 때에 도움이 되는 Mix 태스크들이 존재합니다:
+
+```shell
+mix ecto.create         # 저장소에 공간을 생성합니다
+mix ecto.drop           # 저장소의 공간을 삭제합니다
+mix ecto.gen.migration  # 저장소의 새로운 마이그레이션을 생성합니다
+mix ecto.gen.repo       # 새로운 저장소를 생성합니다
+mix ecto.migrate        # 저장소의 마이그레이션을 실행합니다
+mix ecto.rollback       # 저장소의 마이그레이션을 취소(Rollback)합니다
+```
+
+## 마이그레이션
+
+마이그레이션을 생성하는 가장 좋은 방법은 `mix ecto.gen.migration <name>` 태스크입니다. ActiveRecord를 사용해본 적이 있다면 무척 친숙할 것입니다.
+
+사용자 테이블의 마이그레이션을 확인해봅시다:
+
+```elixir
+defmodule ExampleApp.Repo.Migrations.CreateUser do
+  use Ecto.Migration
+
+  def change do
+    create table(:users) do
+      add :username, :string, unique: true
+      add :encrypted_password, :string, null: false
+      add :email, :string
+      add :confirmed, :boolean, default: false
+
+      timestamps
+    end
+
+    create unique_index(:users, [:username], name: :unique_usernames)
+  end
+end
+```
+
+초기 상태에서 Ecto는 자동으로 증가하는 기본키 `id`를 생성합니다. 이 예제에서는 표준적인 `change/0` 콜백을 사용하고 있습니다만, Ecto에는 보다 세밀한 제어가 필요한 경우를 위해서 `up/0`과 `down/0`에도 지원하고 있습니다.
+
+`timestamps`를 마이그레이션에 추가하면 추측하신 대로, `created_at`과 `updated_at`을 생성하고 관리합니다.
+
+이 새로운 마이그레이션을 적용하려면 `mix ecto.migrate`를 실행해주세요.
+
+마이그레이션의 더 자세한 정보는 Ecto 문서의 [Ecto.Migration](http://hexdocs.pm/ecto/Ecto.Migration.html#content)를 참고해주세요.
+
+## 모델
+
+마이그레이션이 생성되었으므로 이제 모델로 넘어갑시다. 모델은 스키마, 헬퍼 메소드, 그리고 changeset을 정의합니다. changeset에 대해서는 뒤에서 다룹니다.
+
+우선 마이그레이션을 위한 모델이 어떤 것인지 확인해보죠:
+
+```elixir
+defmodule ExampleApp.User do
+  use Ecto.Schema
+  import Ecto.Changeset
+
+  schema "users" do
+    field :username, :string
+    field :encrypted_password, :string
+    field :email, :string
+    field :confirmed, :boolean, default: false
+    field :password, :string, virtual: true
+    field :password_confirmation, :string, virtual: true
+
+    timestamps
+  end
+
+  @required_fields ~w(username encrypted_password email)
+  @optional_fields ~w()
+
+  def changeset(user, params \\ :empty) do
+    user
+    |> cast(params, @required_fields, @optional_fields)
+    |> unique_constraint(:username)
+  end
+end
+```
+
+모델 내에서 정의된 스키마는 마이그레이션에 기술했던 것과 밀접하게 표현됩니다. 여기에서는 데이터베이스의 필드 이외에도 2개의 가상 필드가 추가되어 있습니다. 가상 필드는 데이터베이스에는 저장되지 않습니다만, 검증과 같은 작업에서 도움이 됩니다. 가상 필드에 대해서는 [Changeset](#section-9)에서 살펴봅니다.
+
+## 질의
+
+저장소에 질의하기 위해서는 질의 API를 가져와야 합니다만, 여기에서는 `from/2`만을 가져오는 것으로 충분합니다:
+
+```elixir
+import Ecto.Query, only: [from: 2]
+```
+
+질의 API의 공식 문서는 [Ecto.Query](http://hexdocs.pm/ecto/Ecto.Query.html)에서 찾아보실 수 있습니다.
+
+### 기본
+
+Ecto은 멋진 질의 DSL을 제공하고 있으며, 질의를 이해하기 쉬운 형태로 표현할 수 있습니다. 모든 승인된 계정의 사용자 이름을 검색하는 경우, 다음과 같은 질의를 사용할 수 있을 겁니다:
+
+```elixir
+alias ExampleApp.{Repo,User}
+
+query = from u in User,
+    where: u.confirmed == true,
+    select: u.username
+
+Repo.all(query)
+```
+
+`all/2`뿐 아니라 Repo는 `one/2`나 `get/3`, `insert/2`, `delete/2`를 포함하는 많은 콜백을 제공하고 있습니다. 모든 콜백의 목록은 [Ecto.Repo#callbacks](http://hexdocs.pm/ecto/Ecto.Repo.html#callbacks)에서 확인하실 수 있습니다.
+
+### Count
+
```

Comment:
[this](https://github.com/doomspork/elixir-school/commit/7879a872eb2b9fb54f8a0a64e17f6798904d4ec6#diff-c0e779be9314ec04f7a0f9cdc40fb262) is added just 4 days ago T-T
anyway thanks, I will update this!